include/arch/posix/linker.ld

```diff
@@ -22,22 +22,38 @@
 SECTIONS
  {
 
+	SECTION_PROLOGUE(romstart_snippet,,)
+	{
 /* Located in generated directory. This file is populated by the
  * zephyr_linker_sources() Cmake function.
  */
 #include <snippets-rom-start.ld>
+	}
 
 #include <linker/common-rom.ld>
 
+	SECTION_PROLOGUE(rodata_snippet,,)
+	{
+
 /* Located in generated directory. This file is populated by the
  * zephyr_linker_sources() Cmake function.
  */
 #include <snippets-rodata.ld>
+	} INSERT AFTER .rodata;
```

Comment:
INSERT AFTER applies to the whole `SECTIONS` not to a given section

Reply:
Thanks. I had misread your original proposed patch. Fixed.